src/ast/query.rs

```diff
@@ -1065,9 +1065,12 @@ pub enum JoinConstraint {
 pub struct OrderByExpr {
     pub expr: Expr,
     /// Optional `ASC` or `DESC`
+    /// TODO: Convert this into an Enum (ASC, DESC, Operator)
     pub asc: Option<bool>,
     /// Optional `NULLS FIRST` or `NULLS LAST`
     pub nulls_first: Option<bool>,
+    /// Optional `USING` operator for ordering (PostgreSQL only)
+    pub using: Option<BinaryOperator>,
```

Comment:
This might be OK in our fork, but I'd probably put more thought into this before raising upstream since we'd have to make a breaking change to support ops other than binary ops.

Reply:
Sorry, I don't understand what you mean.

Reply:
If the AST is part of the API, changing `Option<BinaryOperator>` to something else will be a breaking change.